src/runtime/slot-composer.ts

```diff
@@ -102,12 +102,12 @@ export class SlotComposer {
   }
 
   _addSlotConsumer(slot: HostedSlotConsumer) {
-    slot.startRenderCallback = this.arc.pec.startRender.bind(this.arc.pec);
-    slot.stopRenderCallback = this.arc.pec.stopRender.bind(this.arc.pec);
+    slot.startRenderCallback = slot.arc.pec.startRender.bind(slot.arc.pec);
+    slot.stopRenderCallback = slot.arc.pec.stopRender.bind(slot.arc.pec);
     this._consumers.push(slot);
   }
 
-  initializeRecipe(recipeParticles: Particle[]) {
+  initializeRecipe(arc: Arc, recipeParticles: Particle[]) {
```

Comment:
`arc.dispose` calls `slotComposer.dispose` which disposes all slot consumers. 
I think now it needs to pass the arc object and only dispose the consumers of that arc being disposed?

Reply:
Good catch. When I started implementing it I realized I don't know how to do this well. Slot consumers and their provided slot contexts from different arcs can be interwoven.

I.e. outer arc can provide a slot which is consumed by an inner arc, which provides a slot which is then mapped back to an outer arc and consumed again.

How do you dismantle this one arc at a time? You can't.

I don't know what to do in a generic case - but I'm not sure a generic case will ever arrive, we will need to rethink it then.

However for inner arcs - which is what I'm doing, it's enough to dispose an entire slot composer with an outer arc, which is what I will do in the follow up patch. Added a TODO to this patch as a marker that its not 100% done yet.

WDYT?

Reply:
In the simple generic case, we could dismantle an arc, if it doesn't provide slots for other "outer" arcs (only itself and its own inner arcs). We could support this simple case, if we could check and assert this.
Or to begin with, is there maybe a way for us to verify the current inner arcs case - assert that all slot consumers being disposed are necessarily either this arc or its inner arcs?

Reply:
Good idea. I've added an assert to SlotComposer.dispose, to ensure we're not doing something bad there.